lib/src/posthog_flutter_web_handler.dart

```diff
@@ -1,107 +1,180 @@
-// ignore: deprecated_member_use
-import 'dart:js';
+import 'dart:js_interop';
 
 import 'package:flutter/services.dart';
 
-Future<dynamic> handleWebMethodCall(MethodCall call, JsObject context) async {
-  final analytics = JsObject.fromBrowserObject(context['posthog']);
+// Definição da interface JS para o PostHog
+@JS()
+@staticInterop
+class PostHog {}
+
+extension PostHogExtension on PostHog {
+  external JSAny? identify(
+      JSAny userId, JSAny properties, JSAny propertiesSetOnce);
+  external JSAny? capture(JSAny eventName, JSAny properties);
+  external JSAny? alias(JSAny alias);
+  external JSAny? get_distinct_id();
+  external void reset();
+  external void debug(JSAny debug);
+  external JSAny? isFeatureEnabled(JSAny key);
+  external void group(JSAny type, JSAny key, JSAny properties);
+  external void reloadFeatureFlags();
+  external void opt_in_capturing();
+  external void opt_out_capturing();
+  external JSAny? getFeatureFlag(JSAny key);
+  external JSAny? getFeatureFlagPayload(JSAny key);
+  external void register(JSAny properties);
+  external void unregister(JSAny key);
+  external JSAny? get_session_id();
+}
+
+// Acessar o posthog a partir do window
+@JS('window.posthog')
```

Comment:
if the script fails to set `window.posthog`, this will return null, so then `posthog` should be marked as nullable